proforma/proforma_generator.py

```diff
@@ -463,6 +564,62 @@ def fill_alignment(ws, series, current_row, alignment):
     batt_cost_cell = "\'{}\'!B{}".format(inandout_sheet_name, current_row)
     make_attribute_row(ws, current_row)
 
+    if (chp.size_kw or 0) > 0:
+        #replicated logic from reo/process_results.py
+        total_kw = chp.size_kw or 0
+        total_kwh = chp.year_one_electric_energy_produced_kwh or 0
+        total_runtime = sum(np.array(chp.year_one_electric_production_series_kw or []) > 0) / float(scenario.time_steps_per_hour)
+        # Calculate capital cost from cost curve list
+        cost_list = chp.installed_cost_us_dollars_per_kw or []
+        size_list = chp.tech_size_for_cost_curve or []
+        # CHP size object defined below, but use the chp_size_kw_cell to preserve spreadsheet referencing
+        chp_size = total_kw
+        if len(cost_list) > 1:
+            if chp_size <= size_list[0]:
+                chp_installed_cost_us_dollars = "= {} * {}".format(chp_size_kw_cell, cost_list[0])
+            elif chp_size > size_list[-1]:
+                chp_installed_cost_us_dollars = "= {} * {}".format(chp_size_kw_cell, cost_list[-1])
+            else:
+                for s in range(1, len(size_list)):
+                    if (chp_size > size_list[s-1]) and (chp_size <= size_list[s]):
+                        slope = (cost_list[s] * size_list[s] - cost_list[s-1] * size_list[s-1]) / \
+                                    (size_list[s] - size_list[s-1])
+                        chp_installed_cost_us_dollars = "= {} * {} + ({} - {}) * {}".format(
```

Comment:
@Bill-Becker in this loop would we ever set chp_installed_cost_us_dollars twice? should it be `chp_installed_cost_us_dollars +=`  or maybe there should be a break after we set it?

Reply:
It will never set the cost twice because  of the ">" and "<=". You could introduce a break to avoid possible couple of unnecessary iterations on the loop.